Source/UI/IdentityUI.swift

```diff
@@ -477,6 +488,105 @@ extension IdentityUI {
+            show(input: .byLoginMethod(.password,
+                                       presentingViewController: presentingViewController,
+                                       localizedTeaserText: localizedTeaserText,
+                                       scopes: scopes),
+                 client: client,
+                 completion: completion)
+
+            sharedWebCredentialsController = nil
+        }
+
+        sharedWebCredentialsController = SharedWebCredentialsController { [weak self] result in
+            guard case .success((let username, let password)) = result,
+                  let email = EmailAddress(username),
+                  let self = self else {
+                fallback()
+                return
+            }
+
+            self.identityManager.login(username: .email(email),
+                                       password: password,
+                                       scopes: scopes,
+                                       persistUser: true,
+                                       useSharedWebCredentials: false) { result in
+                guard case .success = result else {
+                    fatalError()
+                    return
+                }
+
+                let user = self.identityManager.currentUser
+
+                func success() {
+                    completion(.success(user: user, persistUser: true))
+                }
+
+                func cancel() {
+                    completion(.cancel)
+                }
+
+                func presentTerms() {
+                    self.identityManager.fetchTerms { result in
+                        guard case .success(let terms) = result else {
+                            cancel()
+                            return
+                        }
+
+                        let configuration = IdentityUIConfiguration(
+                            clientConfiguration: user.clientConfiguration)
+
+                        let navigationController = DismissableNavigationController {
+                            if IdentityUI.updatedTermsCoordinator != nil {
+                                user.logout()
+                                cancel()
+                            }
+                            IdentityUI.updatedTermsCoordinator = nil
+                        }
+
+                        let updatedTermsCoordinator = UpdatedTermsCoordinator(
+                            navigationController: navigationController, configuration: configuration)
+
+                        updatedTermsCoordinator.start(input: .init(currentUser: user, terms: terms)) { output in
+                            IdentityUI.updatedTermsCoordinator = nil
+
+                            navigationController.dismiss(animated: true) {
+                                switch output {
+                                case .success:
+                                    success()
+                                case .cancel:
+                                    cancel()
```

Comment:
Same thing w/ logout

Reply:
The logout is already called inside the `UpdatedTermsCoordinator`